config/abilities/provider_any.rb

```diff
@@ -22,7 +21,7 @@
 
     # Services
     user_accessible_services = user.accessible_services
-    can %i[read show edit update], Service, user_accessible_services.where_values_hash unless user_accessible_services.is_a? ActiveRecord::NullRelation
```

Comment:
cancancan's `read` permission enables `index` and `show` controller action.

`index` is handled seperately, based on the permissions that the user is granted.